pkg/workloadapi/workload.proto

```diff
@@ -0,0 +1,101 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package proto;
+option go_package="pkg/workloadapi";
+
+message Workload {
+  // Name represents the name for the workload. This may be elided as an optimization
+  string name = 1;
+  // Namespace represents the namespace for the workload. This may be elided as an optimization
+  string namespace = 8;
+
+  string canonical_name = 14;
+  string canonical_revision = 15;
+  string workload_type = 16;
+  string workload_name = 17;
+
+  // Address represents the IPv4/IPv6 address for the workload.
+  // This should be globally unique.
+  // This should not have a port number.
+  // TODO: Add network as discriminator
+  string address = 2;
+
+  // The SPIFFE identity of the workload.
+  string identity = 3;
+
+  // The protocol that should be used to connect to this workload.
+  Protocol protocol = 4;
+
+  // The address this workload can be reached at.
+  // If not set, address should be reached directly
+  string gateway_address = 5;
+  string remote_proxy = 11;
+  // Temporary solution:
+  // Set to disable node -> node direct calling
+  bool native_hbone = 13;
+
+  // Virtual IPs defines a set of virtual IP addresses the workload can be reached at,
+  map<string, PortList> virtual_ips = 6;
+
+  // Workload UID.
+  string uid = 7;
+
+  // If true, mTLS will be required
+  bool enforce = 9;
+
+  // Node the workload runs on
+  string node = 10;
+
+  Authorization rbac = 12;
+}
+
+message Authorization {
+  bool enforceTLS = 1;
+  repeated Policy allow = 2;
+  repeated Policy deny = 3;
+}
+
+message Policy {
```

Comment:
I am not sure at all that the full policy in affect for an IP should be sent along with the info. 

This is a case where zproxy can get policies separately, and just have a reference to it. 

However - policy enforcement, at least in NetworkPolicy case, is based on label matching. I would not reinvent this - if the purpose is to reimplement 
NetworkPolicy ( since we are bypassing it ), just use NetworkPolicy and send the labels.

Reply:
Cilium already has xDS server and schema for NetworkPolicies, which I assume is widely deployed.

Reply:
> Cilium already has xDS server and schema for NetworkPolicies, which I assume is widely deployed.

can you share the link for that

Reply:
https://github.com/cilium/proxy/blob/master/cilium/api/npds.proto. Implementation is in the same repo.

Reply:
@zirain https://github.com/cilium/proxy/blob/master/cilium/network_policy.cc#L749 is the main xDS method, with a standard pattern to post to thread locals for every update. It is a SotW which seems to be enough for all Cilium clusters.

Reply:
is it the same as k8s NetworkPolicy, or just another policy using same name ? It seems everything is now using same generic names and are 'standards' because they are sent over XDS,  even if they are very vendor specific.

Reply:
Yes, looks like it's L7 network policy in Cilium. My point still stands, stop calling this "Policy", that's super confusing. Everything is a policy on some workloads.

Reply:
Agreed, we should avoid unqualified broad names. Can't change Cilium name, but we should. 

I think Cilium validates the pattern of sending a custom vendor specific object over XDS - but I would postpone merging any authz related in the proto ( I assume it is not implemented or use in ztunnel right now - most of the fields don't seem to be ). The main question is if ztunnel will implement K8SNetworkPolicy ( with label-based extensions ), subset of Istio Auths, adopt some other standard - or invent yet another one. 

All this is not related to this PR.

Reply:
I don't think it's right to project policy onto an endpoint in any case. It's the wrong delta of change. A policy changes should not trigger 1000s of updates of endpoints.

Reply:
I agree, Kuat. I'll remove from this PR and we can rebuild it differently later